addons/website_sale/models/website_snippet_filter.py

```diff
@@ -127,17 +124,22 @@ def _get_products_latest_sold(self, website, limit, domain, **kwargs):
             ('state', '=', 'sale'),
         ], limit=8, order='date_order DESC')
         if sale_orders:
-            sold_products = [p.product_id.id for p in sale_orders.order_line]
-            products_ids = [id for id, _ in Counter(sold_products).most_common()]
-            if products_ids:
+            if self.env.context.get('hide_variants'):
+                sold_products = Counter(
+                    sol.product_id.product_tmpl_id.product_variant_id
+                    for sol in sale_orders.order_line
+                )
+            else:
+                sold_products = Counter(sol.product_id for sol in sale_orders.order_line)
+            if sold_products:
```

Comment:
At that point, all we're interested in are the `id`s, the order only matters further down, where I maintained sorting by frequency by modifying the `sorted` key:
```diff
-                products = products.sorted(key=lambda p: products_ids.index(p.id))[:limit]
+                products = products.sorted(key=sold_products.get, reverse=True)[:limit]
```
Instead of using an $O(n^2 log n)$ sorting algorithm, it's now $O(n log^2 n)$ by simply checking the element's frequency in the `Counter` object.

The use of `most_common` to create a list sorted by frequency only to then use `index` on that list sort a recordset introduced unneeded complexity for both CPUs & maintainers.

Reply:
Indeed, you are right, thank you for your explanation!